### PR TITLE
Make autoconf 2.70 work without issues

### PR DESCRIPTION
**Pull request by kerolasa**
Merry Christmas. A while back autoconf got new release, and when my laptop received the update I decided to test if util-linux work fine which it did not. This pull request fixes the autoconf upgrade related issues, and has additional small change to suppress a make(1) warning.

### CODE REVIEW

**Comment by zeha** [comment]
Travis failure on Ubuntu precise is:

```
configure.ac:115: error: AC_PROG_CC cannot be called after AM_PROG_CC_C_O
configure.ac:115: the top level
```

**Comment by kerolasa** [comment]
> Travis failure on Ubuntu precise

Thank you for pointing that out @zeha I updated the change, lets see if 9aca76f1ab9b6e12e6b332c2973bbc3ba6ba0f5f works better.

**Comment by karelzak** [comment]
It seems it works as expected with old automaker, so let's merge it :-)

File: configure.ac [hunk 1]
```diff
@@ -109,10 +109,9 @@ AS_IF([test "x$SYSCONFSTATICDIR" = x],
...
 AC_SUBST([sysconfstaticdir])
 
-
+AC_PROG_CC
 AM_PROG_CC_C_O
 AC_PROG_MKDIR_P
-AC_PROG_CC_STDC
 AC_PROG_YACC
 AC_CANONICAL_HOST
 AC_C_CONST
```
Comment (karelzak) [inline configure.ac:117]: Does it mean that these AC_* macros are not necessary? Does it automatically check for bison/flex? (AC_PROG_YACC)?

Comment (kerolasa) [reply configure.ac:117]: This removal is related to following release note item:

    *** AC_PROG_CC_STDC, AC_PROG_CC_C89, AC_PROG_CC_C99 are now obsolete.
    
      Applications should use AC_PROG_CC.

Comment (karelzak) [reply configure.ac:117]: OK, thanks for info.

**Pull request merged**